subroutines/src/pcs/multilinear_kzg/batching.rs

```diff
@@ -80,22 +79,42 @@ where
 
     // \tilde g_i(b) = eq(t, i) * f_i(b)
     let timer = start_timer!(|| format!("compute tilde g for {} points", points.len()));
-    let mut tilde_gs = vec![];
-    for (index, f_i) in polynomials.iter().enumerate() {
-        let mut tilde_g_eval = vec![E::Fr::zero(); 1 << num_var];
-        for (j, &f_i_eval) in f_i.iter().enumerate() {
-            tilde_g_eval[j] = f_i_eval * eq_t_i_list[index];
-        }
-        tilde_gs.push(Arc::new(DenseMultilinearExtension::from_evaluations_vec(
-            num_var,
-            tilde_g_eval,
-        )));
-    }
+    let point_indices = points
```

Comment:
Maybe add a comment that we combine polynomials evaluated at the same point?

Reply:
Sure, added in https://github.com/EspressoSystems/hyperplonk/pull/123/commits/f36bb45196199087d8124a9a0e94493144d6b69e and also simplify other stuff.